src/Microsoft.SqlTools.ServiceLayer/EditData/UpdateManagement/RowCreate.cs

```diff
@@ -96,14 +96,21 @@ public override DbCommand GetCommand(DbConnection connection)
             List<string> inValues = new List<string>();
             List<SqlParameter> inParameters = new List<SqlParameter>();
             List<string> selectColumns = new List<string>();
-            for(int i = 0; i < AssociatedObjectMetadata.Columns.Length; i++)
+            for (int i = 0; i < AssociatedObjectMetadata.Columns.Length; i++)
             {
                 DbColumnWrapper column = AssociatedResultSet.Columns[i];
                 EditColumnMetadata metadata = AssociatedObjectMetadata.Columns[i];
                 CellUpdate cell = newCells[i];
-                
+
                 // Add the output columns regardless of whether the column is read only
-                outClauseColumnNames.Add($"inserted.{metadata.EscapedName}");
+                if (metadata.IsHierarchyId)
```

Comment:
Again, I'm not sure this pattern is scalable. You may want to consider having this as a new utilities class that can convert the columns to identifiers.
Alternatively, this is pretty much just adding `inserted.` to `metadata.ExpressionForSelectStatement`. Removing the check for `IsHierarchyId` in favor of using that property would simplify this code a bit